GitUI/Editor/RichTextBoxXhtmlSupportExtension.cs

```diff
@@ -1146,6 +1146,80 @@ public static string GetPlainText(this RichTextBox rtb, int from, int to)
             return text.ToString();
         }
 
+        public static string GetLink(this RichTextBox rtb, int charIndex)
+        {
+            IntPtr oldMask = rtb.BeginUpdate();
+
+            int nStart = rtb.SelectionStart;
+            int nEnd = rtb.SelectionLength;
+
+            try
+            {
+                //--------------------------------
+                // this is an inefficient method to get text format
+                // but RichTextBox doesn't provide another method to
+                // get something like an array of charformat and paraformat
+                //--------------------------------
+                rtb.Select(charIndex, 1);
+                if (!rtb.IsLink())
+                {
+                    return null;
+                }
+
+                int from = charIndex;
+                while (true)
```

Comment:
How can we sure that this is not endless?

Reply:
`from` is decremented in every iteration.
Though I should change it to `while (from > 0)`.
The same applies to the loop for `to`.

I have not written the NUnit test yet.